libs/checkpoint-sqlite/langgraph/checkpoint/sqlite/__init__.py

```diff
@@ -163,14 +163,15 @@ def cursor(self, transaction: bool = True) -> Iterator[sqlite3.Cursor]:
         Yields:
             sqlite3.Cursor: A cursor for the SQLite database.
         """
-        self.setup()
-        cur = self.conn.cursor()
-        try:
-            yield cur
-        finally:
-            if transaction:
-                self.conn.commit()
-            cur.close()
+        with self.lock:
```

Comment:
should we just move the lock into `self.cursor()` and use it everywhere?

Reply:
that's what I did no?

Reply:
yup, my bad on the first part of the question, but saw a few places where we still explicitly do `async with self.lock, self.conn.execute(` -- think we can just use cursor there